framework/test/unit/modules/interoperability/mainchain/commands/message_recovery.spec.ts

```diff
@@ -436,119 +590,40 @@ describe('Mainchain MessageRecoveryCommand', () => {
-					chainID,
-					swapReceivingAndSendingChainIDs(ccm),
-				);
-			}
-		});
-
-		it('should throw when beforeRecoverCCM of ccMethods of the ccm fails', async () => {
-			// Assign & Arrange
-			const method = ({
-				beforeRecoverCCM: jest.fn(() => {
-					throw new Error('beforeRecoverCCM Error');
-				}),
-				MODULE_NAME_INTEROPERABILITY,
-			} as unknown) as BaseCCMethod;
-
-			command['interoperableCCMethods'].set(MODULE_NAME_INTEROPERABILITY, method);
-
-			// Assert
-			await expect(command.execute(commandExecuteContext)).rejects.toThrow(
-				'beforeRecoverCCM Error',
-			);
-		});
-
-		it('should throw when terminated chain outbox does not exist', async () => {
-			// Assign & Arrange
-			const { chainID } = transactionParams;
-
-			when(terminatedOutboxAccountMock.has).calledWith(chainID).mockResolvedValue(false);
-
-			// Assert
-			await expect(command.execute(commandExecuteContext)).rejects.toThrow(
-				'Terminated outbox account does not exist',
-			);
-		});
-
-		it('should not add CCM to outbox when sending chain of the CCM does not exist', async () => {
-			// Assign & Arrange & Act
-			chainAccountStoreMock.has.mockResolvedValue(false);
-
-			await command.execute(commandExecuteContext);
-
-			// Assert
-			expect.assertions(ccms.length);
-			for (const _ of ccms) {
-				expect(command['internalMethod'].addToOutbox).not.toHaveBeenCalled();
-			}
-		});
-
-		it('should not add CCM to outbox when sending chain of the CCM is not live', async () => {
-			// Assign & Arrange & Act
-			for (const ccm of ccms) {
-				const chainID = ccm.sendingChainID;
-
-				when(command['internalMethod'].isLive as never)
-					.calledWith(chainID)
-					.mockResolvedValue(false as never);
-			}
-
-			await command.execute(commandExecuteContext);
+		it('should proceed without error if the sending chain is the mainchain', async () => {
+			ccms = [
+				{
+					nonce: BigInt(0),
+					module: MODULE_NAME_INTEROPERABILITY,
+					crossChainCommand: CROSS_CHAIN_COMMAND_NAME_REGISTRATION,
+					sendingChainID: CHAIN_ID_MAINCHAIN,
+					receivingChainID: utils.intToBuffer(3, 4),
+					fee: BigInt(1),
+					status: 1,
+					params: Buffer.alloc(0),
+				},
+			];
 
-			// Assert
-			expect.assertions(ccms.length);
-			for (const _ of ccms) {
-				expect(command['internalMethod'].addToOutbox).not.toHaveBeenCalled();
-			}
+			commandExecuteContext = createCommandExecuteContext(ccms);
+			await expect(command.execute(commandExecuteContext)).resolves.toBeUndefined();
 		});
 
-		it('should not add CCM to outbox when sending chain of the CCM is not active', async () => {
-			// Assign & Arrange & Act
-			for (const ccm of ccms) {
-				const chainID = ccm.sendingChainID;
-
-				when(chainAccountStoreMock.get)
-					.calledWith(chainID)
-					.mockResolvedValue({
-						status: -1,
-					} as any);
-			}
-
-			await command.execute(commandExecuteContext);
+		it('should proceed without error if the sending chain is not the mainchain', async () => {
```

Comment:
same as above, 
should call forwardRecovery ? and also check forwardRecovery to have been called with right params?

Reply:
also check that terminatedOutboxAccount.set is called or it has been set if in memory calling `.has`? same below